src/docfx/lib/git/GitUtility.cs

```diff
@@ -118,6 +118,46 @@ public static Task Reset(string cwd, string branch)
         public static Task<string> HeadRevision(string cwd)
            => ExecuteQuery(cwd, "rev-parse HEAD", TimeSpan.FromMinutes(3));
 
+        /// <summary>
+        /// Retrieve original URL
+        /// </summary>
+        public static Task<string> GetOriginalUrl(string cwd)
+            => ExecuteQuery(cwd, $"config --get remote.origin.url", TimeSpan.FromMinutes(1));
+
+        /// <summary>
+        /// Retrieve local branch name
+        /// </summary>
+        public static Task<string> GetLocalBranch(string cwd)
+            => ExecuteQuery(cwd, $"rev-parse --abbrev-ref HEAD", TimeSpan.FromMinutes(1));
+
+        /// <summary>
+        /// Retrieve head commit id for local branch
+        /// </summary>
+        public static Task<string> GetLocalBranchCommitId(string cwd)
+            => ExecuteQuery(cwd, $"rev-parse HEAD", TimeSpan.FromMinutes(1));
+
+        /// <summary>
+        /// Retrieve permanent git URL
+        /// </summary>
+        /// <param name="repo">Git repo info</param>
+        /// <param name="path">Path relative to root path of <paramref name="repo"/></param>
+        public static string GetGitPermaLink(GitRepoInfo repo, string path)
```

Comment:
These could go to lib/github/. git != github

Reply:
Moved. Will do some refactor when we support other git hosts.

---
In reply to: [194685485](https://github.com/dotnet/docfx/pull/2876#discussion_r194685485) [](ancestors = 194685485)